examples/text-to-sql/src/data-ingestion.tsx

```diff
@@ -0,0 +1,80 @@
+import * as gensx from "@gensx/core";
+import {
+  DatabaseProvider,
+  useDatabase,
+  useDatabaseStorage,
+} from "@gensx/storage";
+
+// Database initialization component
+const DatabaseInitializer = gensx.Component<{}, string>(
+  "DatabaseInitializer",
+  async () => {
+    const DB_NAME = "baseball";
+
+    const storage = useDatabaseStorage();
+    let currentDatabases = await storage.listDatabases();
+
+    // If the database doesn't exist, populate it
+    // Otherwise, just return a message confirming it exists.
+    if (!currentDatabases.includes(DB_NAME)) {
```

Comment:
Not the most bulletproof method as it relies on ordering. Another suggestion might be: 

- useDatabase
- run the `create table if not exists` (this is idempotent and safe to do every time)
- query the table to find the number of rows. 
- if zero rows, run the insert statement.

Reply:
sounds good, i will update to this